packages/firebase_remote_config/lib/src/remote_config.dart

```diff
@@ -25,17 +25,24 @@ class RemoteConfig extends ChangeNotifier {
   RemoteConfigSettings _remoteConfigSettings;
 
   DateTime get lastFetchTime => _lastFetchTime;
+
   LastFetchStatus get lastFetchStatus => _lastFetchStatus;
+
   RemoteConfigSettings get remoteConfigSettings => _remoteConfigSettings;
 
-  static Completer<RemoteConfig> _instanceCompleter = Completer<RemoteConfig>();
+  @visibleForTesting
+  static Completer<RemoteConfig> instanceCompleter = Completer<RemoteConfig>();
 
   /// Gets the instance of RemoteConfig for the default Firebase app.
   static Future<RemoteConfig> get instance async {
-    if (!_instanceCompleter.isCompleted) {
-      _instanceCompleter.complete(await _getRemoteConfigInstance());
+    if (!instanceCompleter.isCompleted) {
+      try {
+        instanceCompleter.complete(await _getRemoteConfigInstance());
+      } on StateError catch (error) {
+        if (error.message != 'Future already completed') rethrow;
```

Comment:
It seems like we could just check if the `instanceCompleter` is complete at this point, instead of checking the exception in a fragile way.

Reply:
I have rewritten the code to use an if statement.